lib/std/Progress.zig

```diff
@@ -53,50 +60,84 @@ initial_delay_ns: u64 = 500 * std.time.ns_per_ms,
 
 done: bool = true,
 
-/// Protects the `refresh` function, as well as `node.recently_updated_child`.
+/// Protects the `refresh` function, as well as `Node` attributes.
 /// Without this, callsites would call `Node.end` and then free `Node` memory
 /// while it was still being accessed by the `refresh` function.
 update_mutex: std.Thread.Mutex = .{},
 
-/// Keeps track of how many columns in the terminal have been output, so that
+/// Keeps track of how many rows in the terminal have been output, so that
 /// we can move the cursor back later.
+rows_written: usize = undefined,
+
+/// Keeps track of how many cols in the terminal have been output, so that
+/// we can apply truncation if needed.
 columns_written: usize = undefined,
 
+/// Stores the current max width of the terminal.
+/// If not available then 0.
+max_columns: usize = undefined,
+
 /// Represents one unit of progress. Each node can have children nodes, or
 /// one can use integers with `update`.
 pub const Node = struct {
     context: *Progress,
     parent: ?*Node,
     name: []const u8,
     unit: []const u8 = "",
-    /// Must be handled atomically to be thread-safe.
-    recently_updated_child: ?*Node = null,
+    /// Depth of the Node within the tree
+    node_tree_depth: usize = undefined,
+    /// Must be handled using `update_mutex.lock` to be thread-safe.
+    children: [presentable_children]?*Node = [1]?*Node{null} ** presentable_children,
     /// Must be handled atomically to be thread-safe. 0 means null.
     unprotected_estimated_total_items: usize,
     /// Must be handled atomically to be thread-safe.
     unprotected_completed_items: usize,
 
+    const presentable_children = 10;
+
+    /// Push this `Node` to the `parent.children` stack of the provided `Node` (insert at first index). Thread-safe
+    /// Returns void if `parent` is null
+    fn tryPushToParentStack(self: *Node, target_node: *Node) void {
+        if (target_node.parent) |parent| {
```

Comment:
This always returns void.
```suggestion
    fn tryPushToParentStack(self: *Node, target_node: *Node) void {
        const parent = target_node.parent orelse return;
```